plugin.video.wakanim/resources/lib/wakanim.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Wakanim - Watch videos from the german anime platform Wakanim.tv on Kodi.
+# Copyright (C) 2017 MrKrabat
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+
+import xbmc
+import xbmcgui
+import xbmcplugin
+
+import cmdargs
+import login
+import netapi
+import list
+
+
+def main():
+    """Main function for the addon
+    """
+    args = cmdargs.parse_args()
+
+    # check if account is set
+    username = args._addon.getSetting("wakanim_username")
+    password = args._addon.getSetting("wakanim_password")
+
+    # set country
+    args._country = args._addon.getSetting("country")
+    if args._country == "0":
+        args._country = "de"
+    elif args._country == "1":
+        args._country = "fr"
+    elif args._country == "2":
+        args._country = "sc"
+    else:
+        args._country = "de"
+
+    if (username == "") or (password == ""):
+        # open addon settings
+        args._addon.openSettings()
+        return False
+    else:
+        # login
+        success = login.login(username, password, args)
+        if success == True:
+            # list menue
+            xbmcplugin.setContent(int(sys.argv[1]), "tvshows")
+            check_mode(args)
+        else:
+            # login failed
+            xbmc.log("[PLUGIN] %s: Login failed" % args._addonname, xbmc.LOGERROR)
+            xbmcgui.Dialog().ok(args._addonname, args._addon.getLocalizedString(30040))
+            return False
+
+
+def check_mode(args):
+    """Run mode-specific functions
+    """
+    try:
+        mode = args.mode
+    except:
```

Comment:
Don't use bare `except` unless you really know what you are doing. You should test against specific exceptions.